doc/guides/contributing/pull-requests.md

```diff
@@ -251,7 +251,9 @@ And on Windows:
 > vcbuild test
 ```
 
-(See the [running tests][] section of Building guide for more details.)
+(This code might take more than an hour to execute on a PC, want to run a part
+of tests quickly or for more details see the [running tests][] section of
+Building guide.)
```

Comment:
```suggestion
Running tests can take a long time. To run a subset of the test suite, see the
[running tests][] section of the Building guide.
```

Reply:
Agree with you there is no need for too much explanation here. I used the `-j4` argument at run time, but it still takes more than an hour.So I thought it might be a good idea to add a numerical estimated time.

Reply:
Please tell me what you think and finish the PR
@Trott

Reply:
I don't know what's the typical experience (and I'm not sure that anyone else does either) so I'm inclined to *not* try to specify a time and go with the suggestion I made above. However, if we think it would be helpful to include a running time, maybe something like this?: "Depending on your hardware, running tests can take an hour or more."

Reply:
Of course I understand that time is hardware dependent, but based on my tests I don't think I can achieve results in less than an hour without a specialized compilation computer.
"Depending on your hardware, running tests might take an hour or more."maybe a good answer

Reply:
```suggestion
Running all tests takes a long time (typically an hour or more). To run a subset of the
test suite, see the [running tests][] section of the Building guide.
```

Reply:
> I don't think I can achieve results in less than an hour without a specialized compilation computer.

The challenge is that it's not clear that the text added here applies mostly to Windows. It doesn't take anywhere near an hour to run on my very-old not-very-powerful macOS laptop and I suspect Linux laptops can run the test suite even faster.

```suggestion
For some configurations, running all tests might take a long time (an hour or
more). To run a subset of the test suite, see the [running tests][] section of the Building guide.
```

Reply:
So the CPU performance is the determining factor? I have heard that the MAC CPU is very good at some calculations.

Reply:
> So the CPU performance is the determining factor? I have heard that the MAC CPU is very good at some calculations.

I'm not sure that it's CPU. I suspect there's a lot more involved. Maybe someone on @nodejs/platform-windows would have a good idea.

In both Node.js itself as well as in V8 and libuv, there is a lot of platform-specific C++ code. So running Node.js on Windows isn't exactly an apples-to-apples comparison with Node.js on POSIX-like operating systems (macOS, Linux, etc.).

One challenge we've had as a project (for at least the entire 6+ years that I've been involved) is that our core developers are overwhelmingly non-Windows devs. There have been efforts to improve the DX on Windows, but they seem to come and go, and I haven't been aware of any efforts there for some time.

Reply:
On my Windows PC, it takes less than 15 minutes to compile and test.

Reply:
I exceeded more than an hour on both my Windows PC and Linux laptops and the laptop took a little longer. Maybe my hardware is too old?

Reply:
How about using build time in CI as the standard?

Reply:
I think the text as it currently stands sufficiently covers things:

> For some configurations, running all tests might take a long time (an hour or
> more). To run a subset of the test suite, see the [running tests][] section of
> the Building guide.

If that works for you, this can land. We have all the right approvals and CI is green.

Reply:
No problem. I think the current version is pretty good